auditor/src/routes/advanced_record_filters.rs

```diff
@@ -0,0 +1,373 @@
+        .build()
+        .fetch_all(pool)
+        .await
+        .map_err(GetRecordError)?;
+
+    let result: Vec<RecordDatabase> = rows
+        .iter()
+        .map(|row| RecordDatabase {
+            record_id: row.try_get("record_id").unwrap(),
+            meta: row.try_get("meta").ok().unwrap_or(None),
+            components: row.try_get("components").ok().unwrap_or(None),
+            start_time: row.try_get("start_time").ok().unwrap_or(None),
+            stop_time: row.try_get("stop_time").ok().unwrap_or(None),
+            runtime: row.try_get("runtime").ok().unwrap_or(None),
+        })
+        .collect();
+
+    result
+        .into_iter()
+        .map(Record::try_from)
+        .collect::<Result<Vec<Record>, anyhow::Error>>()
+}
+
+#[tracing::instrument(name = "Getting one record using record_id", skip(record_id, pool))]
+pub async fn get_one_record(
+    record_id: String,
+    pool: &PgPool,
+) -> Result<Vec<Record>, anyhow::Error> {
+    let mut query = QueryBuilder::new("SELECT a.record_id,
+                      m.meta,
+                      css.components,
+                      a.start_time,
+                      a.stop_time,
+                      a.runtime
+               FROM accounting a
+               LEFT JOIN (
+                   WITH subquery AS (
+                       SELECT m.record_id as record_id, m.key as key, array_agg(m.value) as values
+                       FROM meta as m
+                       GROUP BY m.record_id, m.key
+                   )
+                   SELECT s.record_id as record_id, array_agg(row(s.key, s.values)) as meta
+                   FROM subquery as s
+                   GROUP BY s.record_id
+                   ) m ON m.record_id = a.record_id
+               LEFT JOIN (
+                   WITH subquery AS (
+                      SELECT 
+                          c.id as cid,
+                          COALESCE(array_agg(row(s.name, s.value)::score) FILTER (WHERE s.name IS NOT NULL AND s.value IS NOT NULL), '{}'::score[]) as scores
+                      FROM components as c
+                      LEFT JOIN components_scores as cs
+                      ON c.id = cs.component_id
+                      LEFT JOIN scores as s
+                      ON cs.score_id = s.id
+                      GROUP BY c.id
+                   )
+                   SELECT rc.record_id as id, array_agg(row(c.name, c.amount, sq.scores)::component) as components
+                   FROM records_components AS rc
+                   LEFT JOIN components as c
+                   ON rc.component_id = c.id
+                   LEFT JOIN subquery AS sq
+                   ON sq.cid = rc.component_id
+                   GROUP BY rc.record_id
+               ) css ON css.id = a.id
+           ");
+
+    let is_valid_record_id = ValidName::parse(record_id.clone().to_string());
+    if is_valid_record_id.is_ok() {
+        query.push(format!("WHERE a.record_id = '{}' ", &record_id));
+    }
```

Comment:
Since the structure of the query is fixed (in contrast to the advanced query), can we go back to `sqlx::query!()`? This way we would have compile-time checks for this SQL query.